subprojects/core/src/main/java/org/gradle/api/internal/project/MutableStateAccessAwareProject.java

```diff
@@ -0,0 +1,1088 @@
+    }
+
+    @Override
+    public Project evaluationDependsOn(String path) throws UnknownProjectException {
+        onMutableStateAccess("evaluationDependsOn");
+        return delegate.evaluationDependsOn(path);
+    }
+
+    @Override
+    public void evaluationDependsOnChildren() {
+        onMutableStateAccess("evaluationDependsOnChildren");
+        delegate.evaluationDependsOnChildren();
+    }
+
+    @Override
+    public Map<String, Project> getChildProjectsUnchecked() {
+        return delegate.getChildProjectsUnchecked();
+    }
+
+    @Override
+    public Set<? extends ProjectInternal> getAllprojects(ProjectInternal referrer) {
+        return delegate.getAllprojects(referrer);
+    }
+
+    @Override
+    public void allprojects(ProjectInternal referrer, Action<? super Project> configureAction) {
+        delegate.allprojects(referrer, configureAction);
+    }
+
+    @Override
+    public DynamicObject getInheritedScope() {
+        return delegate.getInheritedScope();
+    }
+
+    @Override
+    public GradleInternal getGradle() {
+        return delegate.getGradle();
+    }
+
+    @Override
+    public LoggingManager getLogging() {
+        return delegate.getLogging();
+    }
+
+    @Override
+    public Object configure(Object object, Closure configureClosure) {
+        return delegate.configure(object, configureClosure);
+    }
+
+    @Override
+    public Iterable<?> configure(Iterable<?> objects, Closure configureClosure) {
+        return delegate.configure(objects, configureClosure);
+    }
+
+    @Override
+    public <T> Iterable<T> configure(Iterable<T> objects, Action<? super T> configureAction) {
+        return delegate.configure(objects, configureAction);
+    }
+
+    @Override
+    public RepositoryHandler getRepositories() {
+        onMutableStateAccess("repositories");
+        return delegate.getRepositories();
+    }
+
+    @Override
+    public void repositories(Closure configureClosure) {
+        onMutableStateAccess("repositories");
+        delegate.repositories(configureClosure);
+    }
+
+    @Override
+    public DependencyHandler getDependencies() {
+        onMutableStateAccess("dependencies");
+        return delegate.getDependencies();
+    }
+
+    @Override
+    public void dependencies(Closure configureClosure) {
+        onMutableStateAccess("dependencies");
+        delegate.dependencies(configureClosure);
+    }
+
+    @Override
+    public DependencyFactory getDependencyFactory() {
+        return delegate.getDependencyFactory();
+    }
+
+    @Override
+    public ProjectEvaluationListener getProjectEvaluationBroadcaster() {
+        return delegate.getProjectEvaluationBroadcaster();
+    }
+
+    @Override
+    public void addRuleBasedPluginListener(RuleBasedPluginListener listener) {
+        delegate.addRuleBasedPluginListener(listener);
+    }
+
+    @Override
+    public void prepareForRuleBasedPlugins() {
+        delegate.prepareForRuleBasedPlugins();
+    }
```

Comment:
I suppose because of [direct](https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java#L1432) mutation on implementation, it can be considered as a mutable state access?